### DISCUSSION TITLE
2022-03-28 weekly planning

### DISCUSSION THREAD

**Discussion by kgodey** [mathesar-foundation/mathesar#1232] [Maintainer discussions]
## About
At the start of each week, we start a discussion covering Mathesar's progress, team member check ins, and general updates. Everyone is welcome to add a new topic to discuss or to comment on existing topics.

## Discussion topics
- [Last Week in Mathesar](https://github.com/centerofci/mathesar/discussions/1232#discussioncomment-2440530)
- [Frontend priorities](https://github.com/centerofci/mathesar/discussions/1232#discussioncomment-2440531)
- [Backend priorities](https://github.com/centerofci/mathesar/discussions/1232#discussioncomment-2440534)
- [Design priorities](https://github.com/centerofci/mathesar/discussions/1232#discussioncomment-2440536)
- [Product priorities](https://github.com/centerofci/mathesar/discussions/1232#discussioncomment-2440537)
- [Comms Assignee schedule](https://github.com/centerofci/mathesar/discussions/1232#discussioncomment-2440538)
- [Dominykas' check in](https://github.com/centerofci/mathesar/discussions/1232#discussioncomment-2453041)
- [Brent's check in](https://github.com/centerofci/mathesar/discussions/1232#discussioncomment-2454537)
- [Pavish's check in](https://github.com/centerofci/mathesar/discussions/1232#discussioncomment-2455707)
- [Ghislaine's check in](https://github.com/centerofci/mathesar/discussions/1232#discussioncomment-2456068)
- [Kriti's check in](https://github.com/centerofci/mathesar/discussions/1232#discussioncomment-2457256)
- [Sean's check in](https://github.com/centerofci/mathesar/discussions/1232#discussioncomment-2461418)
- [Mukesh's check in](https://github.com/centerofci/mathesar/discussions/1232#discussioncomment-2461912)

## Weekly check-in
@centerofci/mathesar-core please post your weekly check-in as a top-level response and edit the discussion topics above to add your check in as a topic. Others are also welcome to respond.

Prompts:
- How did your most recent work go?
- What are you planning to work on soon?
- Do you anticipate any dependencies or blockers that someone else on the team can help you with?

If you'd like to add anything else to your check-in that's not related to the prompts, feel free to do so.

**Comment by kgodey** [comment]
## Last Week in Mathesar

### Feedback Requested
- https://github.com/centerofci/mathesar-wiki/pull/38 (draft)

### Complete
- **Discussions**:
    - https://github.com/centerofci/mathesar/discussions/1184
    - https://github.com/centerofci/mathesar/discussions/1192
    - https://github.com/centerofci/mathesar/discussions/1193
- **Full stack and general work**:
    - https://github.com/centerofci/mathesar/issues/1128
    - https://github.com/centerofci/mathesar/issues/1196
    - https://github.com/centerofci/mathesar/issues/1197
    - https://github.com/centerofci/mathesar/issues/1168
- **Backend work**:
    - https://github.com/centerofci/mathesar/issues/1121
    - https://github.com/centerofci/mathesar/issues/689
    - https://github.com/centerofci/mathesar/issues/555
    - https://github.com/centerofci/mathesar/issues/1177
    - https://github.com/centerofci/mathesar/issues/996
- **Frontend work**:
    - https://github.com/centerofci/mathesar/issues/929
    - https://github.com/centerofci/mathesar/issues/1195
    - https://github.com/centerofci/mathesar/issues/1040
    - https://github.com/centerofci/mathesar/issues/867
    - https://github.com/centerofci/mathesar/issues/930
    - https://github.com/centerofci/mathesar/issues/985
    - https://github.com/centerofci/mathesar/pull/1213

**Comment by kgodey** [comment]
## Frontend priorities

1. Get open PRs merged:
    1. https://github.com/centerofci/mathesar/pull/1155
    1. https://github.com/centerofci/mathesar/pull/1182
    1. https://github.com/centerofci/mathesar/pull/1204
    1. https://github.com/centerofci/mathesar/pull/1211
    1. https://github.com/centerofci/mathesar/pull/1225
    1. https://github.com/centerofci/mathesar/pull/1229
2. Make progress on high-priority issues for [Initial Data Types](https://github.com/centerofci/mathesar/milestone/11) milestone:
    1. #970
    1. #1108
    1. #257 
    1. #775
3. Continue working on other [Initial Data Types](https://github.com/centerofci/mathesar/milestone/11) frontend issues.

@pavish @seancolsen Please comment below with thoughts/concerns.

**Reply by seancolsen** [reply]
Looks good.

**Reply by pavish** [reply]
Sounds good to me

**Comment by kgodey** [comment]
## Backend priorities

1. Get open PRs completed and merged:
    1. https://github.com/centerofci/mathesar/pull/1174
    1. https://github.com/centerofci/mathesar/pull/1205
    1. https://github.com/centerofci/mathesar/pull/1222
    1. https://github.com/centerofci/mathesar/pull/1223
2. Complete remaining backend work for [Initial Data Types](https://github.com/centerofci/mathesar/milestone/11):
    1. @dmos62:
        1. https://github.com/centerofci/mathesar/issues/1100
        1. https://github.com/centerofci/mathesar/issues/1037
    2. @mathemancer
        1. https://github.com/centerofci/mathesar/issues/408 
        2. https://github.com/centerofci/mathesar/issues/411 
        3. https://github.com/centerofci/mathesar/issues/414
        4. https://github.com/centerofci/mathesar/issues/384
        5. https://github.com/centerofci/mathesar/issues/429
        6. https://github.com/centerofci/mathesar/issues/419
        7. https://github.com/centerofci/mathesar/issues/432
    3. @silentninja:
        1. https://github.com/centerofci/mathesar/issues/1105
        1. https://github.com/centerofci/mathesar/issues/1228
    4. Not assigned to core team members:
        1. https://github.com/centerofci/mathesar/issues/1145 (assigned to @ManishShah120)
        1. https://github.com/centerofci/mathesar/issues/1167 (assigned to @ratika-12)
        1. https://github.com/centerofci/mathesar/issues/1214 (assigned to @Cronus1007)
3. Start work on backend work for [Links between Tables](https://github.com/centerofci/mathesar/milestone/47), some ideas:
    1. https://github.com/centerofci/mathesar/issues/943
    2. https://github.com/centerofci/mathesar/issues/944
    3. https://github.com/centerofci/mathesar/issues/973
    4. https://github.com/centerofci/mathesar/issues/972
4. #1008

@dmos62 @mathemancer @silentninja Please comment below with thoughts/concerns.

**Comment by kgodey** [comment]
## Design priorities

1. Complete one round of review for #1065
2. Submit spec for #978 

@ghislaineguerin @ppii775 Please comment below with thoughts/concerns. @ppii775, please let us know if you have any questions about writing the design spec.

**Comment by kgodey** [comment]
## Product priorities

_No product work is planned for this week._

**Comment by kgodey** [comment]
## Comms Assignee schedule

| Week | Person |
|-|-|
| 2022-03-28 to 2022-04-01 | Sean |
| 2022-04-04 to 2022-04-08 | Ghislaine  |
| 2022-04-11 to 2022-04-12 | Kriti |
| 2022-04-13 to 2022-04-15 | Dominykas |
| 2022-04-18 to 2022-04-22 | Brent |
| 2022-04-25 to 2022-04-29 | Kriti |
| 2022-05-02 to 2022-05-06 | Pavish |
| 2022-05-09 to 2022-05-13 | Mukesh |

### Instructions
You can also find the schedule and instructions for the Comms Assignee on the [wiki page](https://wiki.mathesar.org/team/guide/comms-assignee). Please feel free to update the wiki page if you think of anything useful during your shift.

**Comment by dmos62** [comment]
> How did your most recent work go?

I'm learning a lot about the codebase while doing the type logic refactor. And, I'm delaying a lot of the other tasks, like PR reviews. I feel like the focus that gives me is useful, but I'll happily switch tasks if I'm blocking someone.

> What are you planning to work on soon?

I'm currently working through the tests affected by the refactor. It's taking some time. I'll do my best to at least submit for review during this week, before my week off.

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

No.

**Reply by kgodey** [reply]
I don't think you're blocking any contributor PRs and since you're going on vacation at the end of the week, I think it's fine to focus on getting the refactor done over PR reviews. It would be good if you could see #1174 merged this week as well.

**Comment by mathemancer** [comment]
> How did your most recent work go?

Pretty well. I've been getting tests written for grouping and also spending plenty of time on comms for GSoC hopefuls. PR reviews and messaging have taken lots of time, but I think that's okay. I found some irritating details as I wrote tests for ranged grouping, but I'm sorting those out as they arise.

> What are you planning to work on soon?

I should have a working PR in to complete number type grouping soon, and the methods are somewhat reusable for text grouping so I'll move into that. I'm also keeping an eye (and a half) on GSoC comms and PR reviews. I'm trying to stay extra available for checking in with @dmos62 since he's going to be gone next week and is working on a pretty big and complex refactor.

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

Nope.

**Comment by pavish** [comment]
## Pavish's check in

> How did your most recent work go?

Pretty good. I raised the PR for display options groundwork and a small refactoring PR with it. There were a lot more PR reviews and discussions than the previous week, which took some time, but went well. Overall, the week was very productive for me.

> What are you planning to work on soon?

I'm planning on completing:
* https://github.com/centerofci/mathesar/issues/1250
* https://github.com/centerofci/mathesar/issues/257 (Sadly, I had to keep delaying this because of the sheer amount of groundwork required inorder to make things easier for other types and for FK configuration)
* Figure out date & time type implementation.
* I expect a considerable amount of PR reviews and discussions this week as well.

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

None.

**Comment by ghislaineguerin** [comment]
> How did your most recent work go?

Pretty good, feels like we have a good foundation for the rest of the views design. I liked presenting the design during our weekly meeting and having discussions before posting the spec.

> What are you planning to work on soon?

- Review round for #1065 
- Start design work for #466 

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

- None

**Comment by kgodey** [comment]
> How did your most recent work go?

Not great, I didn't have any time for focused work (being out sick didn't help). I'm generally caught up with review and notifications and discussions and so on.

> What are you planning to work on soon?

I have a bunch of meetings and administrative tasks this week (writing job descriptions, etc.). Other than that, I'd like to finish: 
- #1008 
- https://github.com/centerofci/mathesar-wiki/issues/41

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

Nope.

**Comment by seancolsen** [comment]
## Sean's check in

> How did your most recent work go?

- Going okay. I wish I had more time to code though.

> What are you planning to work on soon?

- #775 is my highest-priority task.
- Also on comms this week.

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

- Nope

**Comment by silentninja** [comment]
## Mukesh's check in

> How did your most recent work go?

- Pretty good overall. Happy to have started working on the issues in `Link with tables` milestone. 

> What are you planning to work on soon?

- Fix bugs with display options and get the Money type display options merged. 
- #943 
- #942 

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

- Nope

**Reply by kgodey** [reply]
@silentninja I assume you'll be working on other stuff other than display options too, right?

**Reply by silentninja** [reply]
> Fix bugs with display options and get the Money type display options merged

This is an immediate concern. I have updated the `planning to work` list with the next set of issues I would be working on.
